packages/core/src/request-builder/header-builder/headers-util.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { MapType } from '@sap-cloud-sdk/util';
+
+export function isNullish(x: any): x is null | undefined {
+  return x === null || x === undefined;
+}
+
+export function toHeaderObject(key: string, value: any): MapType<any> {
+  return isNullish(value) ? {} : { [key]: value };
```

Comment:
what happens if the key is nullish?

Reply:
I now also check for keys. Although this is not used anywhere, you are right, if this is used differently in the future, we are safe.